lib/api/controllers/documentController.js

```diff
@@ -89,6 +89,18 @@ class DocumentController {
 
   /**
    * @param {Request} request
+   * @returns {Promise<Boolean>}
+   */
+  exists(request) {
+    assertHasId(request);
+
+    return this.engine.get(request)
+      .then(() => Promise.resolve(true))
```

Comment:
Use `Bluebird` instead of the native promises. Those are far slower than bluebird (will be fixed in future versions of V8)